lua/oil/init.lua

```diff
@@ -910,6 +906,20 @@ local function load_oil_buffer(bufnr)
   adapter.normalize_url(bufname, finish)
 end
 
+local function close_preview_window_if_not_in_oil()
+  local util = require("oil.util")
+
+  if not util.is_oil_bufnr(0) then
+    local preview_win_id = util.get_preview_win()
+    if preview_win_id then
+      local current_win_id = vim.api.nvim_get_current_win()
+      if current_win_id and current_win_id ~= preview_win_id then
+        vim.api.nvim_win_close(preview_win_id, true)
+      end
+    end
+  end
```

Comment:
Not sure why the tests are not passing... Using `vim.schedule` here makes them pass, but I'd rather not do that to avoid any flickers